src/lib.rs

```diff
@@ -1,41 +1,308 @@
+        }
+        msg::reply(
+            IdentityEvent::CheckedClaim {
+                subject,
+                piece_id,
+                status,
+            },
+            0,
+        )
+        .expect("IDENTITY: Error during replying with IdentityEvent::CheckedClaim");
     }
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn init() {
+    let config: InitIdentity = msg::load().expect("Unable to decode InitIdentity");
+    if config.owner_id == ActorId::zero() {
+        panic!("IDENTITY: Owner MUST be non-zero address");
+    }
+    let id_storage = IdentityStorage {
+        owner_id: config.owner_id,
+        piece_counter: 1,
+        ..Default::default()
+    };
+    IDENTITY = Some(id_storage);
+}
+
+#[gstd::async_main]
+async fn main() {
+    let action: IdentityAction = msg::load().expect("Unable to decode IdentityAction");
+    let identity = unsafe { IDENTITY.get_or_insert(Default::default()) };
+    match action {
+        IdentityAction::IssueClaim {
+            issuer,
+            issuer_signature,
+            subject,
+            data,
+        } => identity.issue_claim(issuer, issuer_signature, subject, data),
+        IdentityAction::ClaimValidationStatus {
+            validator,
+            subject,
+            piece_id,
+            status,
+        } => identity.validation_status(validator, subject, piece_id, status),
+        IdentityAction::VerifyClaim {
+            verifier,
+            verifier_signature,
+            subject,
+            piece_id,
+        } => identity.verify_claim(verifier, verifier_signature, subject, piece_id),
+        IdentityAction::CheckClaim {
+            subject,
+            piece_id,
+            hash,
+        } => identity.check_claim(subject, piece_id, hash),
+    }
+}
+
+#[no_mangle]
+extern "C" fn meta_state() -> *mut [i32; 2] {
+    let state: IdentityStateQuery = msg::load().expect("Unable to decode IdentityStateQuery");
+    let identity = unsafe { IDENTITY.get_or_insert(Default::default()) };
+    let reply = match state {
+        IdentityStateQuery::UserClaims(pkey) => {
+            IdentityStateReply::UserClaims(match identity.user_claims.get(&pkey) {
+                None => BTreeMap::new(),
+                Some(claims) => claims.clone(),
+            })
+        }
+        IdentityStateQuery::Claim(pkey, piece_id) => IdentityStateReply::Claim(
+            identity
+                .user_claims
+                .entry(pkey)
+                .or_default()
+                .get(&piece_id)
+                .cloned(),
+        ),
+        IdentityStateQuery::ValidationStatus(pkey, piece_id) => {
+            IdentityStateReply::ValidationStatus(
+                identity
+                    .user_claims
+                    .get(&pkey)
+                    .expect("No such public key")
+                    .get(&piece_id)
+                    .expect("No such piece_id")
+                    .data
+                    .valid,
+            )
```

Comment:
We try to not panic in the `meta_state` function and return default value if smth fails.
I suggest to return `false` on fail:
```suggestion
            let mut status = false;

            if let Some(user_claim) = identity.user_claims.get(&pkey) {
                if let Some(claim) = user_claim.get(&piece_id) {
                    status = claim.data.valid
                }
            }

            IdentityStateReply::ValidationStatus(status)
```
Or return `Option<bool>` with the `None` variant on fail:
```suggestion
            let mut status = None;

            if let Some(user_claim) = identity.user_claims.get(&pkey) {
                status = user_claim.get(&piece_id);
            }

            IdentityStateReply::ValidationStatus(status)
```